language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -687,6 +672,22 @@ impl LibraVM {
     }
 }
 
+fn check_intersection<K: Eq + std::hash::Hash, V: Eq>(
```

Comment:
Could you add a comment to say what this is doing + why?

Reply:
Also: might make sense to add this as a function on `WriteSet` (I know you need a separate one for events, but still).

Reply:
Nit: naming this something like `intersection_matches` might make the meaning of the bool return clearer.